### ISSUE TITLE
The same recipient can be added several times

### ISSUE THREAD

**Issue by Adirelle** [gopasspw/gopass#2831]
### Summary

GPG recipients can be added using key full identifier, short identifier, email address or name. The same recipient can be added more than once using two of those values.

### Steps To Reproduce

1. (Optional: set GOPASS_HOMEDIR and PASSWORD_STORE_DIR to point to empty directories to avoid interference from existing configuration and stores)
2. Create a new root store: `./gopass init`
3. Add a recipient by full key identifier: `./gopass recipients add C7FBF888C77BF4F24B7FC9DFCAC0D760CF02A070`
4. Add the same recipient by short key identifier: `./gopass recipients add 0xCAC0D760CF02A070`
5. Add the same recipient by email adress: `./gopass recipients add john.doe@example.com`
6.  Add the same recipient by name: `./gopass recipients add "Jonh Doe"`
7. Observe that four "recipients" has been added: `./gopass recipients ls`

### Expected behavior

The recipient is added only once.

### Environment

* OS: Ubuntu 22.04 in WSL2
* OS version: `Linux orgrimmar 5.15.146.1-microsoft-standard-WSL2 #1 SMP Thu Jan 11 04:09:03 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`
* gopass Version: `gopass 1.15.11 go1.22.1 linux amd64` (e390f25aa74a888e066d64470cd03fe2f220942f)
* Installation method: from source

**Comment by dominikschulz** [comment]
This is actually intentional. We did have more sophisticated recipient parsing but that didn't work well with some use cases so we removed most of that and delegate most of the recipient handling to the backend (usually GPG).

We try to handle an recipient as a string that's stored in the recipients file (`.gpg-id`) and passed to the backend (GPG) when necessary.

Since we had this feature and had to remove if after a number of bug reports I'm going to mark this as "won't fix".
However if you have a good idea to make this more useful (e.g. provide a warning or so) w/o making the code overly complicated feel free to open a new issue and possibly send a PR.
